integrations/opensearch/docker-compose.yml

```diff
@@ -1,15 +1,46 @@
+version: '3.8'
+
 services:
-  opensearch:
+  opensearch1:
     image: "opensearchproject/opensearch:2.11.0"
+    container_name: opensearch1
     ports:
-      - 9200:9200
-      - 9600:9600
-    restart: on-failure
+      - "9200:9200"  # HTTP
+      - "9600:9600"  # Performance Analyzer
     environment:
       - discovery.type=single-node
+      - cluster.name=opensearch-cluster-1
+      - node.name=opensearch-node-1
       - "ES_JAVA_OPTS=-Xms1024m -Xmx1024m"
+      - bootstrap.memory_lock=true
     healthcheck:
-        test: curl --fail https://localhost:9200/_cat/health -ku admin:admin || exit 1
-        interval: 10s
-        timeout: 1s
-        retries: 10
\ No newline at end of file
+      test: curl --fail https://localhost:9200/_cat/health -ku admin:admin || exit 1
+      interval: 10s
+      timeout: 1s
+      retries: 10
+    networks:
+      - opensearch-net
+
+  opensearch2:
```

Comment:
in general, for OpenSearch, Haystack Document Store == OpenSearch Index

For this reason, I have the impression that this feature can be tested using a single OpenSearch Docker container with 2 indexes.

If possible, having a single container would make the test suite lighter and more maintainable in the long run.